### ISSUE TITLE
Sharepoint REST download returns incomplete content on aspx files and large files

### ISSUE THREAD

**Issue by slavag** [SharePoint/sp-dev-docs#5247]
## Category

- [x] Bug

## Describe the bug
We're using :
/_api/web/GetFileById('<file_guid>')/$value
Or
/_api/web/GetFileByServerRelativeUrl('<file url>')/$value
We see that on many aspx files and on large files, the content that is returned is incomplete.
In the files properties I see one size and when entire stream is read. We see it constantly in Site Pages aspx files, and from time to time on large files. Network issues are eliminated. Some other people are also complaining about this issue :
https://stackoverflow.com/questions/59737754/sharepoint-rest-api-returns-incomplete-content-of-file-during-downloading

## Expected behavior

Returned content length should be exactly the same as in file properties form Sharepoint. 

## Developer environment

REST API

Thanks

**Comment by andrewconnell** [comment]
I can repro this as well...

**Comment by slavag** [comment]
Microsoft guys, any update on this?
This issues is causing to data corruption on migration and backup processes, one of major sharepoint api is buggy. Please take a care

**Comment by slavag** [comment]
@andrewconnell Hi, is this repository monitored at all by Microsoft ? Thanks

**Comment by JeremyKelley** [comment]
I asked around internally and the guidance I got was to use the Microsoft Graph endpoint or the "download.aspx" URL which supports byte range requests for reliable downloads instead of the /$value.

That said, for site pages, are you expecting to back them up by backing up the *.aspx file?  If that's the target scenario you will likely run into issues.  We don't currently have an officially supported way of working with site pages using the APIs today.

**Comment by slavag** [comment]
@JeremyKelley Thanks for the feedback. Few points :
1) Graph API doesn't supports downloading attachments from the list items. Also Graph API as far as I know doesn't returns all lists and document libraries when enumerating drives (this is the way to work with document library). 
2) As part of backup that we provides to our customers we need to backup also aspx files and currently it's an issue.
3) In many cases I see downloaded size much larger than size in metadata on PDF, Doc and not only in aspx. And vice versa as well, when download size is smaller than size in metadata
4) REST API is officially supported API, why not fixing bugs there ? I can understand that for deprecated API  but as far as I know it's not deprecated yet. And it's a bug !!! And as I said this bug is not only for aspx files, but it's not consistent. Graph API has many limitations for very long period and it's doesn't seems progressing.
So, bottom line REST API is buggy and nobody is fixing those bugs, and Graph API is very limited and can't be used.

**Comment by slavag** [comment]
Sharepoint team 2 month opened (reproducible) bug here without any care to be taken of. Very sad.

**Comment by antayirp** [comment]
I can verify that this bug still persists today in June 2022. Our customers have OneNote files bigger than 15 GB in size (don't ask why). I've tried Sharepoint REST API v1, REST API v2, Graph v1 onenote, copy.asmx webservice, vti_bin author.dll, bytes range in header .... nothing works. So frustating.
